python/coinbase-agentkit/coinbase_agentkit/action_providers/wallet/wallet_action_provider.py

```diff
@@ -0,0 +1,73 @@
+from typing import Any
+
+from ...network import Network
+from ...wallet_providers import WalletProvider
+from ..action_decorator import create_action
+from ..action_provider import ActionProvider
+from .schemas import GetBalanceSchema, GetWalletDetailsSchema
+
+
+class WalletActionProvider(ActionProvider[WalletProvider]):
+    """Provides actions for interacting with wallet functionality."""
+
+    def __init__(self):
+        super().__init__("wallet", [])
+
+    @create_action(
+        name="get_wallet_details",
+        description="""
+    This tool will return the details of the connected wallet including:
+    - Wallet address
+    - Network information (protocol family, network ID, chain ID)
+    - Native token balance
+    - Wallet provider name
+    """,
+        schema=GetWalletDetailsSchema
+    )
+    def get_wallet_details(
+        self,
+        args: dict[str, Any]
+    ) -> str:
+        """Get details about the wallet."""
+        try:
```

Comment:
In all actions, we need to validate the args are in the correct format based on the schema